### DISCUSSION TITLE
How to handle 0 or null series data for pie and donut charts?

### DISCUSSION THREAD

**Discussion by pixieltron** [apexcharts/apexcharts.js#2785] [Q&A]
Hi Apexcharts , how would you handle 0 or empty data in pies at present when all series is null or 0 data? 
no visible chart( 1 very thin line  ) pre-requisite's: have legends turned on with there numbers using formatter *thumb

**Update request:**   this should be a classed as an exception if all chart series are 0 or all null then display grey circle or greyed out pie with text awaiting data , 

Thanks 
Josh

**Comment by pixieltron** [comment]
Hi Thanks for moving to discussions does anybody have any current way they handle this

**Comment by cristsr** [comment]
I had a similar problem and I found this solution:

When the component receive null data, I verify it and update the chart with

``` ts
chart.updateOptions({
  series: [1],
  labels: [],
  colors: ['#f0f0f0'],
});
```

Therefore, the chart is displayed with circle 100% filled and with gray color,

Cris

**Reply by albamatti** [reply]
Thanks! And in what part I can put this?

**Comment by pixieltron** [comment]
Wrap it in the same object then loop data in that object , condition if
data is less <= than 0 or data is null else data , on holiday at mo so no
example but it works hope this helps ☺️
